src/Tracker.cpp

```diff
@@ -1895,3 +1968,157 @@ void Tracker::CollectNearestPoints(KeyFrame& kf, std::set<MapPoint*>& spNearestP
+                priorPointCovariance = 1e-8;
+
+            double entropyReduction = EvaluatePoint(this, point, kf, priorPointCovariance, pMeas->nLevel, prevPointEntropy);
+
+            if(!isnan(entropyReduction) ) //if non NAN, todo (adas) need to figure out why nan, probably ill conditioned covariance
+            {
+                totalPreviousEntropy+=prevPointEntropy;
+                totalEntropyReduction +=entropyReduction;
+                totalEntropyPoints++;
+
+            }
+        }
+    }
+
+    mvKeyFrameBuffer.push_back(mpTempMKF); //store the MKF
+    //ROS_INFO("Tracker: Buffering keyframe: %ld", mvKeyFrameBuffer.size());
+    //ROS_INFO("Tracker: Total MKF Entropy Reduction is: %f", totalEntropyReduction);
+
+    score_pair mp; mp.first = totalEntropyReduction; mp.second = mvKeyFrameBuffer.size() - 1;
+    vKeyframeScores.push_back(mp);
+
+    if(totalEntropyReduction > bestBufferKeyframeScore) //if we have a new winner
+    {
+        bestBufferKeyframeScore = totalEntropyReduction;
+        bestBufferKeyframeIndex = mvKeyFrameBuffer.size() - 1;
+    }
+
+
+}
+
+void Tracker::AddNewKeyFrameFromBuffer(int bufferPosition)
+{
+    if( (int)mvKeyFrameBuffer.size() < bufferPosition) //exceeding index
+        return;
+
+    static GVars3::gvar3<int> gvnCrossCamera("CrossCamera", 1, GVars3::HIDDEN|GVars3::SILENT);
+    //note: should check to make sure index doesnt exceed buffer size
+
+    //pull out the keyframe from the buffer
+    MultiKeyFrame* mpTempMKF =  mvKeyFrameBuffer[bufferPosition];
+
+    //set it's status to NOT a bufferkeyframe (it's getting inserted into the map!
+    mpTempMKF->isBufferMKF = false;
+
+    //iterate through all the measurements in the chosen MKF and all all the measurements to the map points
+    //boost::mutex::scoped_lock lock(mMap.mMutex);
+    for(KeyFramePtrMap::iterator it = mpTempMKF->mmpKeyFrames.begin(); it != mpTempMKF->mmpKeyFrames.end(); ++it)
+    {
+        std::string cName = (it->first);
+        KeyFrame* kf = (it->second);
+
+        //std::cout<<"At Keyframe %x " << &kf <<std::endl;
+
+        //iterate through each measurement map in the keyframe
+        std::vector<MapPoint*> ptsToDelete;
+        for(MeasPtrMap::iterator it_mp = kf->mmpMeasurements.begin(); it_mp != kf->mmpMeasurements.end(); ++it_mp)
+        {
+            MapPoint& kf_point = *(it_mp->first);
+
+            //IMPORTANT: check to see if the pointer is still in the map! It may have been trashed by the map maker
+            if(std::count(mMap.mlpPoints.begin(), mMap.mlpPoints.end(), &kf_point)) //is the point still in the map??
+            {
+                //if(kf_point.mMMData.spMeasurementKFs.count(kf)) //if its not in there already, or is bad (bad points taken car of when handed to map maker)
+                //continue;
+
+                //std::cout<<"inserting keyframe " << kf.mCamName << " at point %x " <<  &kf_point <<std::endl;
```

Comment:
@arunxdas can u plz take a look to see if I can remove the commented code. I think the else statement is taking care of the points not in map, so we can remove the commented if?